omod/src/main/java/org/openmrs/module/coreapps/page/controller/conditionlist/AddConditionPageController.java

```diff
@@ -16,10 +16,29 @@
 
 import org.openmrs.ui.framework.page.PageModel;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.openmrs.module.coreapps.CoreAppsConstants;
+import org.openmrs.api.context.Context;
 
 public class AddConditionPageController {
 	
 	public void controller(PageModel model, @RequestParam(value = "returnUrl", required = false) String returnUrl) {
 		model.addAttribute("returnUrl", returnUrl);
+		String criterion = Context.getAdministrationService().getGlobalProperty(CoreAppsConstants.GLOBAL_PROPERTY_CONDITIONS_CRITERIA);
+	    String criterionArray[] = criterion.split(",");
+		String conditionListClasses= "";
+		for(String criteria : criterionArray) 
```

Comment:
Do you need this?
Why don't you just set it as the default value of the global property in config.xml?

Reply:
@lnball  described that global property should be able to take a list of comma separated values (Diagnosis,Symptoms,Findings). By default the value for global property is set to Diagnosis. The above piece of code is needed when user wants to give a comma separated list

Reply:
Will this work for other languages where for instance Diagnosis is called something else?

Reply:
Thank you for pointing this out. I didn't consider that possibility while making this PR. Any suggestions on how it could be done? I have no idea how values of message.properties could be used in java class.

Reply:
Why not just use the concept class uuid?

Reply:
@dkayiwa  Could you please make it more clear?

Reply:
I mean getting rid of that code and just directly pass the loaded global property value to model.addAttribute() because it will be an already comma separated list of uuids.

Reply:
As per my discussion with Ellen, We are adding this feature keeping in view, the clinicians who find it easy to give some label as input for global property rather than the uuid's.

Reply:
Can you share a link to that discussion?

Reply:
@dkayiwa  I'm Sorry I cant provide any link as we discussed it on Uber call

Reply:
The user friendliness can be done by some sort of settings page which displays the name but stores a uuid. Same way a patient is displayed for selection but internally stored as patient id or uuid.

Reply:
I just got a clue on how this could be done. I will test it and update the changes if it works.

Reply:
@dkayiwa now the PR will work for other languages.

Reply:
Can we get rid of all this code? You should only call getGlobalProperty() and pass the returned value to model.addAttribute()

Reply:
But that wouldn't allow the user to give a comma separated list of criteria as input. I feel its better to have this code

Reply:
The user interface could give the user a multiple select widget, and then build a comma separated list to save after user selection. So the AdministrativeService.getGlobalProperty() returns this command separated list of uuids.

Reply:
I have no idea how this could be implemented? How should the design look like? Where should I be adding code?Does the above code create any future issues?

Reply:
All you need to do is this:
String conditionListClasses = Context.getAdministrationService().getGlobalProperty(CoreAppsConstants.GLOBAL_PROPERTY_CONDITIONS_CRITERIA);
model.addAttribute("conditionListClasses", conditionListClasses);

Reply:
But the user won't be giving uuid's as input? So how do I handle that? and how to implement functionality to make the global property dynamic?

Reply:
Do you know the interface where end users enter values for global properties?

Reply:
I have seen the interface but I don't know to which module it belongs to and how it's implemented.

Reply:
You do not need to care about it because it is already part of the reference application

Reply:
I need some clarification on this, if the user is going to give uuid's as values or label's as values which have to be interpreted by the application?

Reply:
If the user gives label's as values, how do I convert it to uuid's without the above java code and maintain the dynamism ?

Reply:
The end user will set the value of that setting to a comma separated list of uuids for concept classes which you supply to this method:
model.addAttribute("conditionListClasses", conditionListClassUuids)

Reply:
**1)**  
![Screenshot (230)](https://user-images.githubusercontent.com/36665698/60685459-d2db2f00-9ec0-11e9-9ff1-c961237b0f1b.png)

**2)**  
![Screenshot (231)](https://user-images.githubusercontent.com/36665698/60685472-e090b480-9ec0-11e9-9de8-1b3c85e40bae.png)

How does the user input look like- 1) or 2) ?

Reply:
The second one.

Reply:
Oh ok, I get it now . Even I initially had same idea as that of your's, but @lnball described that inputs should be given as in screenshot one considering the user flexibility.

Reply:
What should I be doing ? Should I implement it the way you described ?

Reply:
I do not understand why you are asking that question.

Reply:
I updated the PR